### PR TITLE
Fixes issue #5072 - adds additional support to Weaviate

### LINKED ISSUE #5072: Add option to use _additional fields while executing a Weaviate query
### Feature request

Weaviate has the option to pass _additional field while executing a query
https://weaviate.io/developers/weaviate/api/graphql/additional-properties

It would be good to be able to use this feature and add the response to the results. It is a small change, without breaking the API. We can use the kwargs argument, similar to where_filter in the python class weaviate.py.

### Motivation

When comparing and understanding query results, using certainty is a good way.

### Your contribution

I like to contribute to the PR. As it would be my first contribution, I need to understand the integration tests and build the project, and I already tested the change in my local code sample.

### PR DESCRIPTION
**Pull request by jettro**
Implementation is similar to search_distance and where_filter

# adds 'additional' support to Weaviate queries

Fixes #5072

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:
@dev2049

### CODE REVIEW

File: langchain/vectorstores/weaviate.py [hunk 1]
```diff
@@ -201,6 +201,8 @@ def similarity_search_by_text(
         query_obj = self._client.query.get(self._index_name, self._query_attrs)
         if kwargs.get("where_filter"):
             query_obj = query_obj.with_where(kwargs.get("where_filter"))
+        if kwargs.get("additional"):
```
Comment (dev2049) [inline langchain/vectorstores/weaviate.py:204]: does this apply to other search methods as well? (like similarity_search_by_vector)

Comment (jettro) [reply langchain/vectorstores/weaviate.py:204]: Yes, we can use the same mechanism in the function _similarity_search_by_vector_. They have different values that you can use in the additional field. But that is specified in the Weaviate documentation. Not sure if we should replicate that here as well.

I would not use it in the functions similarity_search_with_score and max_marginal_relevance_search_by_vector. They allready use the mechanism with the additional field. They already return a score.

Do you want me to add it to the similarity_search_by_vector method as well?

Comment (dev2049) [reply langchain/vectorstores/weaviate.py:204]: that'd be great!

File: tests/integration_tests/vectorstores/test_weaviate.py [hunk 2]
```diff
@@ -81,6 +81,23 @@ def test_similarity_search_with_metadata_and_filter(
...
+        """Test end to end construction and search with metadata and additional."""
+        texts = ["foo", "bar", "baz"]
+        metadatas = [{"page": i} for i in range(len(texts))]
+        docsearch = Weaviate.from_texts(
+            texts, embedding_openai, metadatas=metadatas, weaviate_url=weaviate_url
+        )
+        output = docsearch.similarity_search(
+            "foo",
+            k=2,
+            additional=["certainty"],
```
Comment (dev2049) [inline tests/integration_tests/vectorstores/test_weaviate.py:97]: is certainty supposed to be in the output? maybe im misunderstanding what with_additional does

Comment (jettro) [reply tests/integration_tests/vectorstores/test_weaviate.py:97]: You are absolutely right. I am trying to run the integration tests myself. But I cannot make it work. Now that you mention it, I think the output check should be like this:

`assert output == [Document(page_content="foo", metadata={"page": 0, "_additional": {"certainty": 0}})]`

I do not know how to get the value for certainty in the test.

Comment (dev2049) [reply tests/integration_tests/vectorstores/test_weaviate.py:97]: as in you can't make the integration test pass or you can't make it run at all? to run you can do
```bash
pytest tests/integration_tests/vectorstores/test_weaviate.py
```

you'll want to spin up local weaviate container, instructions for that at top of integration test file 
```bash
cd tests/integration_tests/vectorstores/docker-compose
docker compose -f weaviate.yml up
```

Comment (jettro) [reply tests/integration_tests/vectorstores/test_weaviate.py:97]: Thanks, learning about poetry at the same time. I managed to get the integration tests running with your tips. Now it keeps telling me he is missing the OpenAI key:

E           ValueError: Error during query: [{'locations': [{'column': 6, 'line': 1}], 'message': 'explorer: get class: vectorize params: vectorize params: vectorize params: vectorize keywords: remote client vectorize: OpenAI API Key: no api key found neither in request header: X-OpenAI-Api-Key nor in environment variable under OPENAI_APIKEY', 'path': ['Get', 'LangChain_47555ec10c224f01abb7b82f7458c0b6']}]

Trying to figure out what to do here. I added the .env file in the integration_tests folder

Comment (dev2049) [reply tests/integration_tests/vectorstores/test_weaviate.py:97]: the integration test requires OpenAI model. if you have an openai api key just set that in the terminal you're running this in with 
```bash
export OPENAI_API_KEY="<api_key>"
```
if you don't have openai key i can just run the test

Comment (jettro) [reply tests/integration_tests/vectorstores/test_weaviate.py:97]: Sorry, I really tried everything I could,  but I cannot fix it. If you can try the test, then I will try to fix it later for my environment.

**Pull request merged**